classes/submission/maps/Schema.php

```diff
@@ -1191,4 +1221,60 @@ protected function getStageFilesBySubmissions(Enumerable $submissions, array $st
             ->filterByFileStages($stageIds)
             ->getMany();
     }
+
+    /***
+     * Get detail about submission issues. Results are keyed by submission ID.
+     */
+    protected function getIssueBySubmissions(Enumerable $submissions): Enumerable
+    {
+        $submissionIds = [];
+        $submissions->each(function (Submission $submission) use (&$submissionIds) {
+            $submissionIds[] = $submission->getId();
+        });
+
+        $publications = Repo::publication()->getCollector()->filterBySubmissionIds($submissionIds)->getMany();
+
+        $issueIds = $publications->map(fn(Publication $publication) => $publication->getData('issueId'))
```

Comment:
I think you can remove most of the code above, the publications are available at the `Submission`  class, through `getData('publications')`